Post features added in master

manitapote opened the pull request.

manitapote committed:
Merge tag '1.4.13'
RoomRent 01

manitapote committed:
testing purpose

manitapote committed:
Formatted the code

Created Post migration, model, table
[*] user specific offer/ask
[ ] Image information in all offer/ask

Offer and Ask Post done
[*] Done saving, retrieving specified posts
[ ] Problem in image url

Ver.txt deleted

manitapote committed:
Post(offer or ask) set, get added Merge tag '1.4.20'

sadhakbj commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,143 @@
+	{
+		$this->post = $post;
+		$this->user = $user;
+		$this->image = $image;
```
indentation

sadhakbj commented on app/Helper/HelperFacade.php:
```diff
@@ -0,0 +1,14 @@
+
+class HelperFacade extends Facade
+{
+	protected static function getFacadeAccessor()
```
When you can easily manipulate Laravel collection what's the use of making this HelperFacade ? Next thing you need not create this HelperFacade instead create helper file and autoload it, so that it can be called from anywhere.

sadhakbj commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,143 @@
+	{
+		$user  = Auth::guard('api')->user()->user;	
+		$posts = $user->userPost;
+			$image  = Helper::pushArray($user->images, 'image_location');
```
I don't support your idea of this Facade

sadhakbj commented on app/Http/Controllers/PostController.php:
```diff
@@ -0,0 +1,143 @@
+		$posts = $this->post->where('offer_or_ask',$choice)->get();
+		
+		foreach ($posts as $post) {
+			$post->user;
```
What does this `$post->user` do

manitapote replied:
gets the author of the post

sadhakbj commented on app/Http/Controllers/UserController.php:
```diff
@@ -13,77 +12,80 @@
     {
         $this->codeMessage = $codeMessage;
         $this->user        = $user;
         $this->device = $device;
```
indentation

sadhakbj commented on app/Http/Controllers/UserController.php:
```diff
@@ -120,20 +121,27 @@ public function login(LoginRequest $request)
     {
         $user = $this->user->whereActivationToken($token)->update([
             'active' => self::ACTIVE,
             'activation_token' => null,
```
indentation

sadhakbj commented on app/Http/Requests/PostRequest.php:
```diff
@@ -0,0 +1,67 @@
+    public function rules()
+    {
+        return [
+            'title' => 'string|required',
```
indentation

manitapote committed:
Post functionality complete
[*] get posts
[*] save posts

manitapote committed:
Complete Post and User functionality
[x] Image Storage changed
[x] ImageController added
[x] Constants.php file added
[x] Image retrieving functionality added

manitapote committed:
Changes according to PR comments

manitapote committed:
Merge tag '1.4.28a'

manitapote committed:
Merge tag '1.4.28'

manitapote committed:
Merge tag '1.4.28.1'
